frontend/utils/notification.js

```diff
@@ -0,0 +1,26 @@
+export function checkNotification (): void {
+  return !!window.Notification
+}
+
+export async function requestNotificationPermission (): Promise<null | string> {
+  if (!checkNotification() || Notification.permission !== 'default') {
+    return null
+  }
```

Comment:
The behavior we want is:

1. If the user blocked notifications on the dashboard, then don't ask again
2. If they blocked notifications on the dashboard, and explicitly request notifications again via the preferences, then ask again

Reply:
```
export async function requestNotificationPermission (force = false): Promise<null | string> {
  if (!checkNotification() || (!force && Notification.permission !== 'default')) {
    return null
  }

  try {
    return await Notification.requestPermission()
  } catch (e) {
    return null
  }
}
```

This could be the solution. Right?